src/matcha_ml/core/core.py

```diff
@@ -368,17 +369,42 @@ def stack_set(stack_name: str) -> None:
     MatchaConfigService.update(stack)
 
 
-def stack_add(module: str) -> None:
+def stack_add(module_type: str, module_flavor: str) -> None:
     """A function for adding a module by name to the stack.
 
     Args:
-        module (str): The name of the module to add.
+        module_type (str): The type of the module to add e.g. 'experiment_tracker'.
+        module_flavor (str): The flavor of module to add e.g. 'mlflow'.
 
     Raises:
         MatchaInputError: if the stack_name is not a valid stack type
         MatchaError: if there are already resources provisioned.
     """
-    ...
+    module_type = module_type.lower()
+    module_flavor = module_flavor.lower()
+
+    if RemoteStateManager().is_state_provisioned():
+        raise MatchaError(
```

Comment:
Don't forget to add tests that cover the exceptions in this function

Reply:
Added